readthedocs/subscriptions/event_handlers.py

```diff
@@ -301,3 +319,83 @@ def customer_updated_event(event, **kwargs):
             organization_slug=organization.slug,
             email=new_email,
         )
+
+
+@djstripe_receiver_gold_membership(
+    [
+        "checkout.session.async_payment_failed",
+        "checkout.session.async_payment_succeeded",
+        "checkout.session.completed",
+        "customer.subscription.updated",
+    ]
+)
+def gold_membership_events(event, **kwargs):
+    # This event has to be handled only on Read the Docs Community,
+    # where we do support Gold membership.
+    if not settings.USE_PROMOS:
+        return
+
+    if event.type == "checkout.session.completed":
+        stripe_subscription_id = event.data["object"]["subscription"]["id"]
+        stripe_subscription = djstripe.Subscription.objects.filter(
+            id=stripe_subscription_id
+        ).first()
+        if not stripe_subscription:
+            log.info("Stripe subscription not found.")
+            return
+
+        log.bind(stripe_subscription=stripe_subscription)
+        stripe_price = stripe_subscription.items.first().price
+        username = event.data["object"]["client_reference_id"]
+        log.bind(user_username=username)
+        mode = event.data["object"]["mode"]
+        if mode == "subscription":
```

Comment:
Looks like this is missing the logic from when mode is `payment`.

Reply:
Yeah, this is on purpose because we don't have one-time donations anymore.